### ISSUE TITLE
Incorrect base fee calculation

### ISSUE THREAD

**Issue by Inphi** [foundry-rs/foundry#5065]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (ad751b2 2023-05-28T00:04:05.718310776Z)

### What command(s) is the bug in?

anvil

### Operating System

Linux

### Describe the bug

The `baseFeePerGas` decreases even when the block is full. To reproduce, run `anvil -b 1`  (set a block time to make it easy to track block headers) and send transactions the use the gas limit.

**Comment by mattsse** [comment]
could you please prepare a repro?

**Comment by Inphi** [comment]
Sure. To reproduce, deploy the following contract repeatedly with a high gas limit.
First run `anvil -b 4`. Setting a block time ensures blocks are always being built and thus the base fee per gas is always adjusted even for empty blocks. Wait for a couple empty blocks until the fee is zero.

Repro.sol:
```solidity
pragma solidity ^0.8.13;

contract Repro {
    constructor() {
        uint256 i = 0;
        while (true) {
            i++;
            if (gasleft() < 100000) {
                return;
            }
        }
    }
}
```

Deploy the above contract repeatedly using the following:
```bash
for x in $(seq 1 1000); do
  forge create --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 Repro --gas-limit 25000000
done
```
This will consume roughly 25 million gas per block. You can confirm this by examining `gasUsed` on incoming blocks.
While the above contract deployments are running, inspect the base fee per gas in incoming blocks. The fee will be stuck at zero.

**Comment by Evalir** [comment]
This was fixed with #5067:
<img width="1327" alt="Screenshot 2023-06-14 at 21 04 18" src="https://github.com/foundry-rs/foundry/assets/26014927/8b008c56-c4e2-4d27-a94f-77a0bf9159d4">
